pkg/util/kernel/proc.go

```diff
@@ -49,3 +55,50 @@ func WithAllProcs(procRoot string, fn func(int) error) error {
 	}
 	return nil
 }
+
+// scanNullString is a SplitFunc for a Scanner that returns each null-terminated string as a token
+func scanNullStrings(data []byte, atEOF bool) (advance int, token []byte, err error) {
+	if atEOF && len(data) == 0 {
+		return 0, nil, nil
+	}
+	if i := bytes.IndexByte(data, '\x00'); i >= 0 {
+		// We have a full null-terminated line.
+		return i + 1, data[0:i], nil
+	}
+	// If we're at EOF, we have a final, non-terminated line. Return it.
+	if atEOF {
+		return len(data), data, nil
+	}
+	// Request more data.
+	return 0, nil, nil
+}
+
+func getEnvVariableFromBuffer(reader io.Reader, envVar string) string {
+	scanner := bufio.NewScanner(reader)
+	scanner.Split(scanNullStrings)
+	for scanner.Scan() {
+		parts := strings.SplitN(scanner.Text(), "=", 2)
+		if len(parts) != 2 {
+			continue
+		}
+
+		if parts[0] == envVar {
+			return parts[1]
+		}
+	}
+
+	return ""
+}
+
+// GetProcessEnvVariable retrieves the given environment variable for the specified process ID, without
+// loading the entire environment into memory. Will return an empty string if the variable is not found.
+func GetProcessEnvVariable(pid int, procRoot string, envVar string) (string, error) {
```

Comment:
are you sure we don't have in the entire repo an existing implementation for that? it is really strange that we are the first one encountering the need to read an env var

Reply:
There are some implementations (one in [servicediscovery](https://github.com/DataDog/datadog-agent/blob/guillermo.julian/eventmonitor-common-process-consumer/pkg/collector/corechecks/servicediscovery/module/envs.go#L181), another in [ptracer](https://github.com/DataDog/datadog-agent/blob/guillermo.julian/eventmonitor-common-process-consumer/pkg/security/ptracer/proc.go#L199) - which also truncates variables, another in [serverless/proc](https://github.com/DataDog/datadog-agent/blob/guillermo.julian/eventmonitor-common-process-consumer/pkg/serverless/proc/proc.go#L51)), but all share two main issues: they're not in any common package, and they load all the environment variables into memory. In our case where we know which variable we want, it doesn't make sense to load everything for every process.